### ISSUE TITLE
Completion in minibuffer: Making inhibit-read-only local to *helm-mode-completion-at-point* while let-bound!

### ISSUE THREAD

**Issue by synic** [syl20bnr/spacemacs#4224]
I often open up spacemacs and type `:e ~/Pro[tab]` to open a file.  Tab completion is hard to see because that message pops up, obscuring the completion.  

How can I make that stop?
#### System Info
- OS: darwin
- Emacs: 24.5.7
- Spacemacs: 0.105.0
- Spacemacs branch: develop (rev. 03c83e1)
- Distribution: spacemacs
- Layers:

``` elisp
(unscroll auto-completion xkcd emacs-lisp git dash yaml github javascript html markdown extra-langs python ruby org osx python django syntax-checking spell-checking version-control lua colors spacemacs-layouts vimscript
          (c-c++ :variables c-c++-enable-clang-support t)
          (shell :variables shell-default-height 30 shell-default-position 'bottom))
```

**Comment by StreakyCobra** [comment]
It would help us to help you if you can provide some more details :-) Have a look [here](https://github.com/syl20bnr/spacemacs/wiki/Debugging#how-to-make-a-great-bug-report).

**Comment by justbur** [comment]
I believe this happens when you try to do tab completion while helm is loading which happens on about a 1 second delay

**Comment by synic** [comment]
It's not while it's loading, I can wait on the home screen forever and it will still happen, *however*, it appears to only happen the first time.

Is there a way do non-helm-completion in the minibuffer?  if I type `e: ~/Pro[tab], it would guess first `~/Projects`, but if I hit tab again, it just brings up the next possible match without helm?  `~/Products`?

**Comment by justbur** [comment]
To clarify, I believe the problem occurs if you execute a command that
enters the minibuffer before helm is loaded. I think if you wait until helm
loads before entering :e you will not see the behavior. Is that correct?

**Comment by synic** [comment]
I can wait as long as I want.  It will still happen.

**Comment by justbur** [comment]
Here's a more direct test. Comment out [this line](https://github.com/syl20bnr/spacemacs/blob/develop/layers/%2Bdistribution/spacemacs-base/packages.el#L472), restart and see if you can reproduce it

**Comment by StreakyCobra** [comment]
@justbur I think your link is (already) outdated. You should link to a specific commit version (pressing `y` on the github page helps, IIRC)

**Comment by justbur** [comment]
Actually I can reproduce this even if helm is not deferred, so I was wrong about that

**Comment by puyo** [comment]
I was seeing this issue this morning, then Spacemacs seemed to update to v.0.105.21 and now the issue seems to have gone away.

**Comment by hansent** [comment]
@synic did you figure out a workaround for this?

**Comment by localhostport80** [comment]
@synic @hansent Yeah I am interested too. That's really annoying and I lose about 2 seconds each time this pops up 😦

**Comment by xguerin** [comment]
@synic @hansent I am interested as well.
